validations/transactWriteItems.js

```diff
@@ -0,0 +1,194 @@
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+            UpdateExpression: {
+              type: 'String',
+            },
+          },
+        },
+        Delete: {
+          type: 'FieldStruct<Delete>',
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+          },
+        },
+        ConditionCheck: {
+          type: 'FieldStruct<ConditionCheck>',
+          children: {
+            TableName: {
+              type: 'String',
+            },
+            ExpressionAttributeValues: {
+              type: 'Map<AttributeValue>',
+              children: 'AttrStruct<ValueStruct>',
+            },
+            ExpressionAttributeNames: {
+              type: 'Map<java.lang.String>',
+              children: 'String',
+            },
+            ReturnValuesOnConditionCheckFailure: {
+              type: 'String',
+              enum: [ 'ALL_OLD', 'NONE' ],
+            },
+            ConditionExpression: {
+              type: 'String',
+            },
+            Key: {
+              type: 'Map<AttributeValue>',
+              notNull: true,
+              children: 'AttrStruct<ValueStruct>',
+            },
+          },
+        },
+      },
+    },
+  },
+}
+
+exports.custom = function (data, store) {
+  var i, request, msg, key
+  for (i = 0; i < data.TransactItems.length; i++) {
+    request = data.TransactItems[i]
+    if (request.Put) {
+      for (key in request.Put.Item) {
+        msg = validations.validateAttributeValue(request.Put.Item[key])
+        if (msg) return msg
```

Comment:
subject to style debate:

```suggestion
        if (msg = validations.validateAttributeValue(request.Put.Item[key]))
          return msg
```

Reply:
Those inline assignments within conditionals are clever but are also imo a bit unusual, and certainly more difficult to debug later. They also blow up `eslint:recommended` on [no-cond-assign](https://eslint.org/docs/latest/rules/no-cond-assign).